### PR TITLE
[batch] Use more precise terminology around instance cost, price and revenue

### PR DESCRIPTION
**Pull request by daniel-goldstein**
The batch code makes copious use of the word "cost", which can cause confusion when considering "whose cost?". This PR attempts to use more precise language (definitions provided by Investopedia):

- Cost is the expense incurred for making a product
- Price is what a customer would pay for a product

Since we are the "company" in question in these definitions, we define what the cloud bills us as the cost of an instance, and what we charge users as the price. Note that the price is higher than the cost because we charge a service fee. This can be a little tricky to get right across the codebase because we as batch operators play both the role as the company to the user and customer to the cloud provider. All of the code changed here though pertains to batch as a company and the definitions should match accordingly.

### CODE REVIEW

File: batch/batch/instance_config.py [hunk 1]
```diff
@@ -94,34 +94,33 @@ def _cost_per_hour_from_resources(resource_rates: Dict[str, float], resources: L
...
         self,
         resource_rates: Dict[str, float],
-        utilized_cores_mcpu: int,
     ) -> float:
-        assert 0 <= utilized_cores_mcpu <= self.cores * 1000
-        if self.job_private:
-            assert utilized_cores_mcpu == self.cores * 1000
-            memory_in_bytes = self.instance_memory()
-        else:
-            memory_in_bytes = self.cores_mcpu_to_memory_bytes(utilized_cores_mcpu)
```
Comment (sjparsa) [inline batch/batch/instance_config.py:117]: I think now the function cores_mcpu_to_memory_bytes can be deleted from the Instance Config classes (this one and the azure and gcp implementations)

**Review by sjparsa — COMMENTED** [review]
I think this is a great change!!

File: batch/batch/instance_config.py (hunk 1 above)
Comment (daniel-goldstein) [reply batch/batch/instance_config.py:117]: Great catch! addressed

**Review by sjparsa — APPROVED** [review]
Looks great!

**Pull request merged**